README.md

```diff
@@ -1,11 +1,83 @@
 # README
-To build use:
-`mvn package`
+
+## Star's K Nearest Neighbors
+
+### ✏️ Description
+
+The Project uses a simple REPL to generate the k nearest neighbors of a star based on their euclidean distance based on
+a given CSV file that contains star information and user's choice of star or coordinates.
+
+### ⚙️ Design
+
+This part of the project (involving star-knn) includes two main classes: Main.java and StarDistance.java. The main
+classes includes the REPL that reads in INPUT data in the format of `stars data/stars/csv_file_name.csv`
+and a query either in the format of `naive_neighbors k x y z` or `naive_neighbors k "star name"`. The file outputs the k
+nearest neighbors of the star based on the name given or the (x,y,z) coordinate. The StarDistance class mainly handles
+forming and calculating distances between stars. This data is stored in hashmaps when a new location/star is queried. It
+has two main methods: `formCoordinate <x> <y> <z>` and `euclideanDistance <coordiate1> <coordinate2>`.
+
+#### Assumptions:
+
+1. There are no duplicate star names.
+2. No star names contain a comma, but could have spaces, or other special characters.
+3. When searching based on coordinates instead of a star name, even if the coordinate queried is the exact same as that
+   of the star's, the returned results will contain the star with the same coordinate. If we are querying the star name,
+   we will not include the star itself in the output. (see comparison between `k_nearest_to96GPSC_num.test`
+   and `k_nearest_to_96GPSC.test`
+   in additional system testing).
+
+#### Implementations
+
```

Comment:
Great approach! Alternatively, to reduce space, instead of creating a separate hashmap for the star ID -> relative distance, you could move the distance calculator to inside a new Star class, and then use a Comparator that calculates the relative distances between all the stars with the queried star/coordinate, and then loop from 0 to k-1 and output k numbers of stars from the sorted list. Hopefully that makes sense!

Reply:
Great thinking! I definitely see the benefit of defining my own comparator in another class that contains these methods. To handle the equidistant stars situation, I would consider defining what is my comparator so I could pick a star randomly if the relative distance is equal.